astro.config.ts

```diff
@@ -44,11 +45,18 @@ export default defineConfig({
                       let matcher = regex.exec(code)
                       if (matcher) {
                         let newCode = code
-                        const path = dirname(relative(root, id)).replace(/\\/g, '/')
+                        const fileRootDir = dirname(id)
+
                         do {
-                          images.push(`${path}/${matcher[2]}`)
-                          newCode = newCode.replaceAll(matcher[1], `${path}/${matcher[2]}`)
-                          newCode = newCode.replaceAll(`(${matcher[2]})`, `(${path}/${matcher[2]})`)
+                          const assetPath = resolve(fileRootDir, matcher[2])
```

Comment:
move this declaration outside the loop: 
```ts
let assetsPath: string
```

Reply:
but this depends on the file from `matcher[2]`

we are resolving `matcher[2]` based on currently processed markdown file's absolute location

Reply:
we don't need a new variable on each iteration, just move the declaration outside the loop, in the loop just assign the value to it

Reply:
add the type, we're using ts: `let assetPath: string`

later I'll add 2 scripts for lint

Reply:
done